src/GRA.Data/Repository/AvatarBundleRepository.cs

```diff
@@ -21,6 +21,47 @@ public class AvatarBundleRepository
         {
         }
 
+        public async Task<List<UserLog>> UserHistoryAsync(int userId)
+        {
+            var userLogs = await _context.UserLogs
+               .AsNoTracking()
+               .Where(_ => _.UserId == userId
+                      && !_.IsDeleted)
+               .OrderBy(_ => _.CreatedAt)
+               .ProjectTo<UserLog>(_mapper.ConfigurationProvider)
+               .ToListAsync();
+
+            foreach (var userLog in userLogs)
+            {
+                if (userLog.BadgeId != null)
+                {
+                    userLog.BadgeFilename = _context.Badges
+                        .AsNoTracking()
+                        .Where(_ => _.Id == userLog.BadgeId)
+                        .SingleOrDefault()
+                        .Filename;
+                }
+            }
+
+            return userLogs;
+        }
+
+        public async Task UpdateHasBeenViewed(int userId, int bundleId)
+        {
+            var userLog = await _context.UserLogs
+            .Where(_ => _.UserId == userId && !_.IsDeleted && _.AvatarBundleId == bundleId)
```

Comment:
Indent these line continuations, ensure that you don't go past 100 characters on a line when you do. It will probably end up being:
```
var userLog = await _context.UserLogs
    .Where(_ => _.UserId == userId 
        && !_.IsDeleted 
        && _.AvatarBundleId == bundleId)
    .ToListAsync();
```